packages/react/src/components/AccountSettings/ChangePassword/ChangePassword.tsx

```diff
@@ -1,53 +1,165 @@
 import React from 'react';
+import isEqual from 'lodash/isEqual';
 
 import { Logger } from 'aws-amplify';
-import { changePassword, translate } from '@aws-amplify/ui';
+import {
+  changePassword,
+  ValidatorSpec,
+  InputEventType,
+  getDefaultConfirmPasswordValidators,
+  getDefaultPasswordValidators,
+  runFieldValidators,
+  translate,
+} from '@aws-amplify/ui';
 
 import { useAuth } from '../../../internal';
 import { View, Flex } from '../../../primitives';
+import { FormValues, BlurredFields, ValidationError } from '../types';
 import {
-  DefaultCurrentPassword,
+  DefaultPasswordField,
   DefaultError,
-  DefaultNewPassword,
   DefaultSubmitButton,
 } from './defaultComponents';
 import { ChangePasswordProps } from './types';
-import { FormValues } from '../types';
 
 const logger = new Logger('ChangePassword');
 
+const getIsDisabled = (
+  formValues: FormValues,
+  validationError: ValidationError
+): boolean => {
+  const { currentPassword, newPassword, confirmPassword } = formValues;
+
+  const hasEmptyField = !currentPassword || !newPassword || !confirmPassword;
```

Comment:
This isn't super important or anything but for readability it would be a bit easier if we just early return from this line since we know we don't need to evaluate the lengths of the validation errors below

Reply:
[b38848d](https://github.com/aws-amplify/amplify-ui/pull/2900/commits/b38848da27d1979f72e70aaf79b8a59eb97ba27a)